pkgs/servers/mail/dovecot/default.nix

```diff
@@ -66,6 +66,9 @@ stdenv.mkDerivation rec {
       url = "https://salsa.debian.org/debian/dovecot/-/raw/debian/1%252.3.19.1+dfsg1-2/debian/patches/Support-openssl-3.0.patch";
       hash = "sha256-PbBB1jIY3jIC8Js1NY93zkV0gISGUq7Nc67Ul5tN7sw=";
     })
+  ] ++ lib.optionals stdenv.isDarwin [
+    # fix timespec calls
+    ./timespec.patch
```

Comment:
this upstream commit fixes the issue.
https://github.com/dovecot/core/pull/211

https://github.com/dovecot/core/commit/0a6198c690c063f05934b4981793e1ddb7ca24b3
https://github.com/dovecot/core/commit/3294f9074a1c6f2b64f956a21dea3b8b51b33aaf

Reply:
Nice spot. So this is probably going to land in the next official stable release. Do you when this is going to happen.

Reply:
i thought 23.11 went out today https://github.com/NixOS/nixpkgs/issues/258640

Reply:
I ment the next official dovecot release. :)

Reply:
Thx for looking. Looks like future major release will be released from  `main`: https://dovecot.org/pipermail/dovecot-news/2022-February/000472.html
So this could mean the commits you mentioned might not land until 2.4. So it might make sense to include the patch for the time being.